pkg/operator/controller/ingress/status.go

```diff
@@ -775,10 +828,58 @@ func computeLoadBalancerStatus(ic *operatorv1.IngressController, service *corev1
 			Message: message,
 		})
 	}
-
 	return conditions
 }
 
+// computeLoadBalancerProgressingStatus returns the LoadBalancerProgressing
+// conditions for the given ingress controller which is used later to determine
+// the ingress controller's Progressing status.
+func computeLoadBalancerProgressingStatus(ic *operatorv1.IngressController, service *corev1.Service, platform *configv1.PlatformStatus) operatorv1.OperatorCondition {
+	// Compute the IngressControllerLoadBalancerProgressingConditionType condition for the LoadBalancer
+	if ic.Status.EndpointPublishingStrategy.Type == operatorv1.LoadBalancerServiceStrategyType {
+		switch {
+		case ic.Status.EndpointPublishingStrategy.LoadBalancer == nil:
+			return operatorv1.OperatorCondition{
+				Type:    IngressControllerLoadBalancerProgressingConditionType,
+				Status:  operatorv1.ConditionUnknown,
+				Reason:  "LoadBalancerStatusIncomplete",
+				Message: "status.endpointPublishingStrategy.loadBalancer is not set.",
+			}
+		case service == nil:
+			return operatorv1.OperatorCondition{
+				Type:    IngressControllerLoadBalancerProgressingConditionType,
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "NoService",
+				Message: "LoadBalancer Service not created.",
+			}
+		}
+		wantScope := ic.Status.EndpointPublishingStrategy.LoadBalancer.Scope
```

Comment:
Whoops, I missed this comment, sorry about that, but good catch. Technically, it doesn't matter because of the `returns` in the previous cases, but should stay consistent.
Fixed.